src/main/scala/mesosphere/marathon/core/volume/providers/DVDIProvider.scala

```diff
@@ -0,0 +1,245 @@
+package mesosphere.marathon.core.volume.providers
+
+import com.wix.accord.{ Validator, _ }
+import com.wix.accord.dsl._
+import mesosphere.marathon.core.volume._
+import mesosphere.marathon.state._
+import org.apache.mesos.Protos.{ ContainerInfo, Environment, Volume => MesosVolume }
+
+import scala.collection.JavaConverters._
+
+/**
+  * DVDIProvider (Docker Volume Driver Interface provider) handles persistent volumes allocated
+  * by a specific docker volume driver plugin. This works for both docker and mesos containerizers,
+  * albeit with some limitations:
+  *   - only a single volume driver per container is allowed when using the docker containerizer
+  *   - docker containerizer requires that referenced volumes be created prior to application launch
+  *   - mesos containerizer only supports volumes mounted in RW mode
+  */
+protected[volume] case object DVDIProvider
+    extends AbstractPersistentVolumeProvider("dvdi")
+    with OptionSupport {
+
+  import org.apache.mesos.Protos.Volume.Mode
+
+  abstract class ProviderOption(
+    override val name: String,
```

Comment:
you do not need to override the fields with just the same thing. Just get rid of `override val`. If possible, even get rid of the complete inheritance here -- only the validation is really polymorphic.